TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/BotActivity.java

```diff
@@ -29,4 +37,41 @@ protected void onCreate(Bundle savedInstanceState) {
         directions.setAdapter(adapter);
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.bot_menu, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.usb_devices_list_menu_item:
+                showConnectedDevices();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    private void showConnectedDevices() {
+        UsbManager manager = (UsbManager) getSystemService(Context.USB_SERVICE);
+        HashMap<String, UsbDevice> devices = manager.getDeviceList();
+        StringBuilder builder = new StringBuilder();
+        if (devices.isEmpty()) {
+            builder.append(getString(R.string.no_connected_devices));
+        } else {
+            for (UsbDevice device : devices.values()) {
+                builder.append(
+                        getString(R.string.usb_device_name_vendor_product,
+                                  device.getDeviceName(),
+                                  device.getVendorId(),
+                                  device.getProductId()
+                        )
+                );
+            }
```

Comment:
Ah good point! I had it before using the StringBuilder and I forgot to add it back 👍